### ISSUE TITLE
False positive `unused-import` when import is used via a parent module

### ISSUE THREAD

**Issue by macdjord** [PyCQA/pylint#8501]
### Bug description

`test.py`:
```python
"""Test"""

import prettyprinter as _pp
import prettyprinter.doctypes


_: _pp.doctypes.Doc
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:4:0: W0611: Unused import prettyprinter.doctypes (unused-import)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

### Expected behavior

`prettyprinter.doctypes` was accessed on line 7, so no warning should be emitted for it.

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
```

### OS / Environment

Win10
